app/views/exercises/_editor.html.slim

```diff
@@ -24,13 +24,29 @@
       = render('editor_button', data: {:'data-placement' => 'top', :'data-toggle' => 'tooltip', :'data-container' => 'body'}, icon: 'fa fa-trophy', id: 'assess', label: t('exercises.editor.score'), title: t('shared.tooltips.shortcut', shortcut: 'ALT + s')) unless @embed_options[:disable_score]
       - unless hide_rfc_button
         = render('editor_button', icon: 'fa fa-comment', id: 'requestComments', label: t('exercises.editor.requestComments'), title: t('exercises.editor.requestCommentsTooltip'))
+
     - @files.each do |file|
       - file.read_only = true if @embed_options[:read_only]
       = render('editor_frame', exercise: exercise, file: file)
-    #autosave-label
-      = t('exercises.editor.lastsaved')
-      span
-      button style="display:none" id="autosave"
+
+    #statusbar.d-flex.justify-content-between
+      div
+        - if !@embed_options[:disable_download] && @exercise.hide_file_tree?
```

Comment:
I couldn't agree more and totally see your points!

Because refactorings are not so easy, these views don't get touched too often (even though they would deserve it). The general code quality of the project is improvable and I am aware of many places I would like to refactor; especially within the `/exercise` views, controllers, and model(s).

As there is not any presenter in the app yet, I will check out some details and make a proposal regarding the integration and start using them. For now, I'll add it to my personal to-do list.

Reply:
> Because refactorings are not so easy, these views don't get touched too often (even though they would deserve it).

That's the crucial point IMO. And this is why the presenters and other refactorings would be the most valuable in this part of the codebase.

Have fun whenever you get around to it! Happy to help out if I can. I love this tricky challenge of pulling strings in complex parts of the code without breaking things, to extract small, clean parts, and see it gradually improve over time.

Reply:
> That's the crucial point IMO. And this is why the presenters and other refactorings would be the most valuable in this part of the codebase.

I know that and the nice feeling of a clean code base providing the same features. For this PR, I will resolve the conversation and postpone the refactoring to a new ticket.